crates/ika-core/src/dwallet_mpc/mpc_session.rs

```diff
@@ -321,6 +461,42 @@ impl DWalletMPCSession {
         Ok(())
     }
 
+    /// Build the vector of messages from all the attempts.
+    ///
+    /// Take messages from the first attempt until the start of the second attempt, from the second until the
+    /// start of the third, and so on.
+    fn build_input_mpc_messages(&self) -> Vec<HashMap<PartyID, MPCMessage>> {
+        Self::build_input_mpc_messages_static(&self.attempts)
+    }
+
+    /// A static version of [`Self::build_input_mpc_messages_static`] for testing purposes.
+    fn build_input_mpc_messages_static(
+        attempts: &Vec<Attempt>,
+    ) -> Vec<HashMap<PartyID, MPCMessage>> {
+        let mut messages = vec![];
+        let mut last_processed_round = 0;
+
+        for i in 0..attempts.len() {
+            if i + 1 < attempts.len() {
```

Comment:
```suggestion
            if i < attempts.len() - 1 {
```
The last index you get is `attempts.len() - 1`